src/repligit/asyncio/client.py

```diff
@@ -0,0 +1,90 @@
+from typing import List
+
+import aiohttp
+
+from repligit.asyncio.parse import decode_lines, iter_lines
+from repligit.parse import generate_fetch_pack_request, generate_send_pack_header
+
+
+async def ls_remote(url: str, username: str = None, password: str = None):
+    """Get commit hash of remote master branch, return SHA-1 hex string or
+    None if no remote commits.
+    """
+
+    url = f"{url}/info/refs?service=git-upload-pack"
+    auth = aiohttp.BasicAuth(username, password) if username or password else None
+    async with aiohttp.ClientSession(auth=auth) as session:
+        async with session.get(url, raise_for_status=True) as resp:
+            lines = decode_lines(iter_lines(resp, encoding="utf-8"))
+            service_line = await anext(lines)
+            assert service_line == "# service=git-upload-pack"
+
+            # `async for` inside `dict()` not supported so no dict comprehension
+            result = {}
+            async for line in lines:
+                if line:
+                    sha, ref = line.split()
+                    result[ref] = sha
+            return result
```

Comment:
```suggestion
            async for line in lines:
                if not line:
                    continue
                sha, ref = line.split()
                result[ref] = sha
            return result
```

Let's flip this condition to exit early and reduce the indentation level to make it a bit easier to read.